pkg/prometheus/agent/statefulset.go

```diff
@@ -155,47 +154,24 @@ func makeStatefulSetSpec(
 ) (*appsv1.StatefulSetSpec, error) {
 	cpf := p.GetCommonPrometheusFields()
 
-	pImagePath, err := operator.BuildImagePath(
+	pImagePath, err := operator.BuildImagePathForAgent(
 		operator.StringPtrValOrDefault(cpf.Image, ""),
 		operator.StringValOrDefault("", c.PrometheusDefaultBaseImage),
```

Comment:
(nit) is it the same, right?
```suggestion
		c.PrometheusDefaultBaseImage,
```

Reply:
There're other places like this too. Like `operator.StringPtrValOrDefault(a.Spec.Image, "")` I'll fix this in a separate PR which we can merge fast, since #6703 depends on this.

Reply:
Ah no. They're not similar. #6703 is for string pointer, here is the string value.